### PR TITLE
build: constrain the dependencies to major versions

### PR DESCRIPTION
**Pull request by manuelcgallucci**
Completed requierments.txt with the libraries and the versions.

### CODE REVIEW

**Review by PierreLeveau — CHANGES_REQUESTED** [review]
the dependencies are in `setup.py`. All the dependencies already listed should be constrained to the major version, like `"tenacity>=8.0.0,<9.0.0"` for example.

File: docs/requirements.txt [hunk 1]
```diff
@@ -1,3 +1,31 @@
...
+charset-normalizer==3.1.0
+click==8.1.3
+cuid==0.4
+ffmpeg-python==0.2.0
+filelock==3.11.0
+future==0.18.3
+gql==3.4.0
+graphql-core==3.2.3
+idna==3.4
+kili==2.131.1
```
Comment (p-desaintchamas) [inline docs/requirements.txt:10]: Not sure this should be hardcoded, else doc will never be generated for new versions of the sdk right ?

**Comment by manuelcgallucci** [comment]
Added dependencies in setup.py, only limited to mayor releases, for libraries with minor releases only, the following was defined, example:
"cuid==0.4" in pip freeze, then in setup.py -> "cuid>=0.4,<0.5"

File: setup.py [hunk 2]
```diff
@@ -5,57 +5,57 @@
...
 
 install_requires = [
-    "pandas",
-    "click",
-    "requests",
-    "tabulate",
-    "tenacity",
-    "tqdm",
-    "typeguard<3.0.0",
+    "pandas>=1.0.0",
```
Comment (PierreLeveau) [inline setup.py:8]: here and below: missing upper bound `"pandas>=1.0.0,<2.0.0"`

**Review by PierreLeveau — CHANGES_REQUESTED** [review]

**Review by Jonas1312 — CHANGES_REQUESTED** [review]
no need to change docs/requirements.txt

**Review by Jonas1312 — CHANGES_REQUESTED** [review]
the docs/requirements.txt is not needed anymore?

why do we lock some packages to minor versions and not major?

the package "filelock" is gone, need to merge/rebase and fix conflicts

**Comment by PierreLeveau** [comment]
> the docs/requirements.txt is not needed anymore?
> 
> why do we lock some packages to minor versions and not major?
> 
> the package "filelock" is gone, need to merge/rebase and fix conflicts

indeed, the doc requirements should be restored.
There can be minor version locking if a feature we use is only present starting from a given minor version. This is also a common practice to lock the minor when the version is <1.0

**Review by Jonas1312 — APPROVED** [review]

**Review by theodu — APPROVED** [review]

**Review by PierreLeveau — APPROVED** [review]

**Pull request merged**